### PR TITLE
Add OCIRU package and CI jobs

### PR DESCRIPTION
**Pull request by chouquette**
### What does this PR do?

This PR introduces a new package type for the remote updater payload, and adds jobs to build and deploy those

### Motivation

https://datadoghq.atlassian.net/browse/APL-2745

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: release.json [hunk 1]
```diff
@@ -7,8 +7,8 @@
     },
     "nightly": {
         "INTEGRATIONS_CORE_VERSION": "master",
-        "OMNIBUS_SOFTWARE_VERSION": "fcd374ae322102d4fcd91a0e7dd4ec5726cdcfb7",
-        "OMNIBUS_RUBY_VERSION": "40bf4965ea054d0967a7cb93ddc1681225df1b4f",
+        "OMNIBUS_SOFTWARE_VERSION": "master",
+        "OMNIBUS_RUBY_VERSION": "chouquette/fix_oci_packaging",
```
Comment (Pythyu) [inline release.json:11]: 🐒 Nitpick :
Reminder to swap back to non-dev branch

File: release.json [hunk 2]
```diff
@@ -27,8 +27,8 @@
     },
     "nightly-a7": {
         "INTEGRATIONS_CORE_VERSION": "master",
-        "OMNIBUS_SOFTWARE_VERSION": "fcd374ae322102d4fcd91a0e7dd4ec5726cdcfb7",
-        "OMNIBUS_RUBY_VERSION": "40bf4965ea054d0967a7cb93ddc1681225df1b4f",
+        "OMNIBUS_SOFTWARE_VERSION": "master",
+        "OMNIBUS_RUBY_VERSION": "chouquette/fix_oci_packaging",
```
Comment (Pythyu) [inline release.json:31]: 🐒 Nitpick :
Same as above

**Review by Pythyu — APPROVED** [review]
Overall look good to me 👍
I'd like to have a bit more context on the new OCIRU packages, it looks like it's just compressing a redhat build into a tarball ?

**Comment by chouquette** [comment]
> I'd like to have a bit more context on the new OCIRU packages, it looks like it's just compressing a redhat build into a tarball ?

That's basically it, It's an XZ archive with a few metadata, notably a checksum of all the included files.
For the sake of correctness, it doesn't have to be built on RHEL nor DEB specifically, it's just that the initial implementation chose to use a RHEL runner, but there's nothing distribution specific in the resulting artifact. As such, the remote updater payload can be used to update an agent on either distribution

**Comment by chouquette** [comment]
The complete spec can be found here: https://github.com/opencontainers/runtime-spec
It might also be worth mentioning that OCIRU stands for Open Container Initiative Remote Updater

**Comment by Pythyu** [comment]
I see makes sense ! Thanks for the clarification 😃

File: release.json (hunk 1 above)
Comment (chouquette) [reply release.json]: Done :)

File: release.json [hunk 3]
```diff
@@ -99,4 +99,4 @@
     "dca-1.9.0": {
         "SECURITY_AGENT_POLICIES_VERSION": "v0.3"
     }
-}
```
Comment (paulcacheux) [inline release.json:102]: I think we can remove this change

Comment (chouquette) [reply release.json:102]: Indeed it's fairly useless and probably pulled in more code owners, sorry about that

**Pull request merged**